### PR TITLE
Add multiple build targets, for umd and es5.

### PR DESCRIPTION
**Pull request by Hinton**
Based upon https://github.com/alexjoverm/typescript-library-starter

VS Code and Visual Studio uses the commonjs library for their IntelliSense, while webpack prefers the es5 version.

As an added bonus, documentation gets generated using typedoc.

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,2 +0,0 @@
-tsconfig.json
-src
```
Comment (kspearrin) [inline .npmignore]: we want to publish the src files to npm too?

File: .npmignore [hunk 2]
```diff
@@ -1,7 +0,0 @@
-.vs
```
Comment (kspearrin) [inline .npmignore]: Why do we not want to ignore this stuff?

File: rollup.config.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import sourceMaps from 'rollup-plugin-sourcemaps';
+
+const pkg = require('./package.json');
+
+const libraryName = 'index';
+
+export default {
+  input: `dist/es/${libraryName}.js`,
+  output: [
+    { file: pkg.main, name: camelCase(libraryName), format: 'umd' },
```
Comment (kspearrin) [inline rollup.config.ts]: commelCase lib seems unnecessary?

File: .npmignore (hunk 2 above)
Comment (Hinton) [reply .npmignore]: The `files` attribute in `package.json` tells npm to only include the `dist` directory.

File: rollup.config.ts (hunk 3 above)
Comment (Hinton) [reply rollup.config.ts]: Probably, I just copied the example.

**Pull request merged**